File: backend/app/schemas/game.py
from typing import Optional

from pydantic import BaseModel


class GameRead(BaseModel):
    id: int
    title: Optional[str]
    release_date: Optional[str]
    image_url: Optional[str]
    discount_percent: Optional[int]
    original_price: Optional[float]
    final_price: Optional[float]
    discount_amount: Optional[float]
    product_url: Optional[str]

    class Config:
        orm_mode = True

File: frontend/src/types/game.ts
//src/types/game.ts
export type Game = {
  id: number;
  title: string | null;
  release_date: string | null;
  image_url: string | null;
  discount_percent: number | null;
  original_price: number | null;
  final_price: number | null;
  discount_amount: number | null;
  product_url: string | null;
};